account_bank_statement_import_coda/wizard/account_bank_statement_import_coda.py

```diff
@@ -0,0 +1,218 @@
+
+    def get_st_information_msg(self, line):
+        """ This method returns formatted informations from an information
+        line
+        """
+        return ('Information with Ref. ' + line.ref +
+                '\n' +
+                'Communication: ' + line.communication +
+                '\n')
+
+    def get_st_line_note_msg(self, line):
+        """This method returns a formatted note from line information
+        """
+        note = []
+        if line.counterparty_name:
+            note.append(_('Counter Party') + ': ' +
+                        line.counterparty_name)
+        if line.counterparty_number:
+            note.append(_('Counter Party Account') + ': ' +
+                        line.counterparty_number)
+        if line.counterparty_address:
+            note.append(_('Counter Party Address') + ': ' +
+                        line.counterparty_address)
+        if line.communication:
+            note.append(_('Communication') + ': ' +
+                        line.communication)
+        return note and '\n'.join(note) or None
+
+    def get_st_line_name(self, line, globalisation_dict):
+        """
+        This method must return a valid name for the statement line
+        The name is the statement communication if exists or
+        the communication of the related globalisation line if exists or
+        '/'
+        """
+        name = line.communication
+        if not name and line.ref_move in globalisation_dict:
+            name = globalisation_dict[line.ref_move].communication
+        return name or '/'
+
+    def get_st_line_vals(self, line, globalisation_dict):
+        """
+        This method must return a dict of vals that can be passed to create
+        method of statement line in order to record it. It is the
+        responsibility of every parser to give this dict of vals,
+        so each one can implement his own way of recording the lines.
+            :param:  line: a dict of vals that represent a line of
+            result_row_list
+            :return: dict of values to give to the create method of
+            statement line,
+                     it MUST contain at least:
+                {
+                    'name':value,
+                    'date':value,
+                    'amount':value,
+                    'ref':value,
+                }
+        """
+        amount = line.transaction_amount
+        if line.transaction_amount_sign == AmountSign.DEBIT:
+            amount = - amount
+        return {'name': self.get_st_line_name(line, globalisation_dict),
+                'date': line.entry_date or datetime.datetime.now().date(),
+                'amount': amount,
+                'ref': line.ref,
+                'partner_name': line.counterparty_name or None,
+                'account_number': line.counterparty_number or None,
+                'note': self.get_st_line_note_msg(line),
+                'unique_import_id': line.ref + line.transaction_ref,
```

Comment:
Is line.ref a good element to make a unique line id? Perhaps the year + paper_seq_number + transaction_ref makes a more unique identifier?